rest/xattr_upgrade_test.go

```diff
@@ -224,8 +227,10 @@ func TestCheckForUpgradeFeed(t *testing.T) {
 	assert.NoError(t, err, "Error writing doc w/ xattr")
 	rt.WaitForSequence(1)
 
+	// Use non admin channels since it is in the channel cache. Using admin using * channel which will not find this document.
+
 	// Attempt to update the documents via Sync Gateway.  Should trigger checkForUpgrade handling to detect metadata in xattr, and update normally.
-	changes := rt.PostChangesAdmin("/{{.keyspace}}/_changes", "{}")
+	changes := rt.GetChanges("/{{.keyspace}}/_changes", user)
```

Comment:
I am confused about what this test is testing for here. But changing this to get changes by user means this will just return the user doc. If that is what the test is trying to do (I'm really not sure myself) then we should alter the above comments to make it clear that the documnet we expect to see on changes here is the user doc and not `TestCheckForUpgrade` doc. 

If this test is trying to see the document `TestCheckForUpgrade` added above this changes call doesn't return that I think because the documents metadata doesn't actually have channel A allocated to it `"channels": [null]`.

Reply:
Skipped with a comment, as per discussion with Adam. This doesn't work with GSI and that's expected. This test was for a rolling upgrade from non xattrs->xattrs and the non xattr nodes will already have this value in the cache.

This is an edge case that we won't fix so skipped the test.